src/Programs/data/ProgramsMetadata.ts

```diff
@@ -24,17 +25,24 @@ function bitFlumeRequirements() {
   };
 }
 
+// Throws if program name is invalid.
+function assignProgramName(name: string) {
+  const programName = resolveProgramFilePath(name);
+  if (!programName) throw new Error("Error while trying to initialize program metadata.");
+  return programName;
+}
+
 interface IProgramCreationParams {
   key: string;
-  name: string;
+  name: ProgramFilePath;
   create: IProgramCreate | null;
   run: (args: string[], server: BaseServer) => void;
 }
 
 export const programsMetadata: IProgramCreationParams[] = [
   {
     key: "NukeProgram",
-    name: "NUKE.exe",
+    name: assignProgramName("NUKE.exe"),
```

Comment:
Couldn't you leave these as strings, and do the `assignProgramName` once, later?

Reply:
Yeah potentially. I think the other option is basically how I'm doing it in Literatures, where the constructor takes in a parent type (in that case Enum, but for programs it would just be string) and then applies FilePath typechecking on top.

Once I'm done with the remaining autocompletion edits, I will try to make the type assertions for the Lits/Msg/Programs look nicer and all be handled similarly to each other.